lib/modules/platform/gitlab/index.ts

```diff
@@ -482,10 +482,13 @@ export async function getBranchStatus(
   }
   logger.debug(`Got res with ${branchStatuses.length} results`);
 
-  const mrStatus = (await getBranchPr(branchName))?.headPipelineStatus;
-  if (!is.undefined(mrStatus)) {
+  const mr = await getBranchPr(branchName);
+  if (mr?.sha !== mr?.headPipelineSha) {
+    logger.debug(
+      'Merge request head pipeline has different sha to commit, assuming merged results pipeline',
+    );
     branchStatuses.push({
-      status: mrStatus as BranchState,
+      status: mr?.headPipelineStatus as BranchState,
```

Comment:
```suggestion
      status: mr.headPipelineStatus as BranchState,
```
should not be undefined here , validate existence